Core/Src/main.c

```diff
@@ -87,14 +89,41 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // Initializing Values
+  const uint8_t transmit_val[3] = {0x1, 0x8, 0x0};
+  const uint8_t receive_val[3] = {};
+  const uint16_t max_adc = 1023;
+  const uint16_t min_pwn_val = 3277; // 5% of counter period (65535)
```

Comment:
It is a good practice to capitalize every letter in a constant variable, so when people refer to it, people know that is variable is actually just a constant number